airflow/providers/microsoft/azure/log/wasb_task_handler.py

```diff
@@ -180,15 +180,20 @@ def wasb_write(self, log: str, remote_log_location: str, append: bool = True) ->
             the new log is appended to any existing logs.
         :type append: bool
         """
-        if append and self.wasb_log_exists(remote_log_location):
-            old_log = self.wasb_read(remote_log_location)
-            log = '\n'.join([old_log, log]) if old_log else log
-
         try:
-            self.hook.load_string(
-                log,
-                self.wasb_container,
-                remote_log_location,
-            )
+            if append and self.wasb_log_exists(remote_log_location):
+                self.hook.append_string(
+                    log,
+                    self.wasb_container,
+                    remote_log_location
+                )
+            else:
+                self.hook.load_string(
+                    log,
+                    self.wasb_container,
+                    remote_log_location,
+                    blob_type = 'AppendBlob',
+                    overwrite=True
+                )
```

Comment:
This looks like a breaking change. Take, for example, you have a BlockBlob log already in the container and you want to append to it, won't it be rejected since this is AppendBlob?

Reply:
@ephraimbuddy Maybe handling that case in the exception and converting an BlockBlob to AppendBlob? What do u think?

Reply:
or simply, use the read-overwrite method based on BlobkBlob....

Reply:
> or simply, use the read-overwrite method based on BlockBlob

I prefer the above with the `overwrite` being an argument to the wasb_write method so it's not hardcoded.
Also, see https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-block-blobs--append-blobs--and-page-blobs